### PR TITLE
Nmallick1/kusto copilot v0

### PR DESCRIPTION
**Pull request by nmallick1**
## Overview
Adding Kusto Copilot for Analytics

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## Solution description
This PR enables Kusto co-pilot for analytics

## How Can This Be Tested? <span>&#128269;</span>
Try using the feature from Tools menu

## Checklist <span>&#9989;</span>
- [X] I have looked over the diffs.
- [X] I have run the linter to catch any errors.
- [X] There are no errors from my changes on this branch.
- [X] I have ensured that my changes support accessibility and can be accessed with the keyboard alone
- [X] I have requested review on this PR.

## Screenshots After Fix (if appropriate):
image

### CODE REVIEW

File: AngularApp/projects/diagnostic-data/src/lib/components/openai-chat-container/openai-chat-container.component.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    var browserType = PortalUtils.getBrowserType();
+    var url = window.location.href;
+    let debugInfo = `${newline}============ Debug Info ============${newline}`;
+    debugInfo += `Browser: ${browserType}${newline}`;
+    debugInfo += `Last User Message Id: ${this.lastMessageIdForFeedback}${newline}`;
+    debugInfo += `Url: ${url}${newline}`;
+
+
+    body = `${body}${newline}${newline}${newline}${debugInfo}`;
+    link = `mailto:applensv2team@microsoft.com?subject=${subject}&body=${body}`;
```
Comment (ShekharGupta1988) [inline AngularApp/projects/diagnostic-data/src/lib/components/openai-chat-container/openai-chat-container.component.ts:145]: nit: can the email be configurable too ?

**Review by ShekharGupta1988 — APPROVED** [review]

**Review by rekhaswaminathan — APPROVED** [review]

**Review by SharmaAjay19 — APPROVED** [review]

File: AngularApp/projects/diagnostic-data/src/lib/components/openai-chat-container/openai-chat-container.component.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  // Utilities
+  ChatAlignment,
+  ChatModel,
+  ResponseTokensSize,
+  APIProtocol
+
+} from "../../../public_api";
+import { HttpClient } from '@angular/common/http';
+import { KeyValuePair } from '../../models/common-models';
+import { PortalUtils } from 'projects/applens/src/app/shared/utilities/portal-util';
```
Comment (SharmaAjay19) [inline AngularApp/projects/diagnostic-data/src/lib/components/openai-chat-container/openai-chat-container.component.ts:19]: Import from relative path or maybe move the utility to the diagnostic-data project?

Comment (nmallick1) [reply AngularApp/projects/diagnostic-data/src/lib/components/openai-chat-container/openai-chat-container.component.ts]: Removed it. This was causing build failures so used an alternative instead.

File: AngularApp/projects/diagnostic-data/src/lib/components/openai-chat-container/openai-chat-container.component.ts (hunk 1 above)
Comment (nmallick1) [reply AngularApp/projects/diagnostic-data/src/lib/components/openai-chat-container/openai-chat-container.component.ts]: Changes added

**Pull request merged**